scripts/dumpapp

```diff
@@ -53,19 +40,26 @@ def main():
   except KeyboardInterrupt:
     sys.exit(1)
 
-def read_frames(conn):
+def read_frames(sock):
   while True:
     # All frames have a single character code followed by a big-endian int
-    code = read_input(conn, 1, 'code')
-    n = struct.unpack('!L', read_input(conn, 4, 'int4'))[0]
+    code = read_input(sock, 1, 'code')
+    n = struct.unpack('!L', read_input(sock, 4, 'int4'))[0]
 
     if code == b'1':
       if n > 0:
-        sys.stdout.buffer.write(read_input(conn, n, 'stdout blob'))
+        sys.stdout.buffer.write(read_input(sock, n, 'stdout blob'))
     elif code == b'2':
       if n > 0:
-        sys.stderr.buffer.write(read_input(conn, n, 'stderr blob'))
+        sys.stderr.buffer.write(read_input(sock, n, 'stderr blob'))
         sys.stderr.buffer.flush()
+    elif code == b'_':
```

Comment:
Consider forwarding stdin unconditionally.  What you have will be sufficient for most plugins, but now that we have a full-duplex transport, it seems unnecessarily restrictive.